rule.php

```diff
@@ -22,9 +22,18 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-defined('MOODLE_INTERNAL') || die();
+defined ( 'MOODLE_INTERNAL' ) || die ();
 
-require_once($CFG->dirroot . '/mod/quiz/accessrule/accessrulebase.php');
+global $CFG;
+
+// This work-around is required until Moodle 4.2 is the lowest version we support.
+if (class_exists('\mod_quiz\local\access_rule_base')) {
+    // Use aliases at class_loader level to maintain compatibility.
+    \class_alias(mod_quiz\local\access_rule_base::class, quiz_access_rule_base::class);
```

Comment:
It's better for plugin to be written for latest version, with some additions for previous version, so if dropping some version support some day we only remove unneded conditions.

Not shure if it will work, but please try
```
if ($CFG->version < 2023042400.00) { // Moodle 4.2.
     require_once($CFG->dirroot . '/mod/quiz/accessrule/accessrulebase.php');
     \class_alias(quiz_access_rule_base::class, mod_quiz\local\access_rule_base::class);
     \class_alias(quiz::class, mod_quiz\quiz_settings::class);
}
```
and change code below to use mod_quiz\local\access_rule_base and mod_quiz\quiz_settings